gql/parser_test.go

```diff
@@ -827,3 +835,267 @@ func TestParseVariablesError7(t *testing.T) {
 		t.Error("Expected type for variable $d")
 	}
 }
+
+func TestParseFilter_simplest(t *testing.T) {
+	query := `
+	query {
+		me(_uid_:0x0a) {
+			friends @filter() {
+				name @filter(namefilter())
+			}
+			gender @filter(eq()),age @filter(neq("a", "b"))
```

Comment:
I suppose you are referring to name @filter inside friends @filter.

I see that as a useful construct. You apply a filter on friends to select friends you want to view. And then a friend might have 1000 names, so maybe you want to add another filter on the name predicate. I thought that was included in the design in https://discuss.dgraph.io/t/filters-in-graphql/831/26 ?